scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -523,9 +523,13 @@ object TextLine {
     new TextLine(p, sm, textEncoding)
 }
 
-class TextLine(p: String, override val sinkMode: SinkMode, override val textEncoding: String) extends FixedPathSource(p) with TextLineScheme {
+class TextLine(p: String, override val sinkMode: SinkMode, override val textEncoding: String)(
+    implicit val tset: TupleSetter[String]
```

Comment:
can we remove this and instead do:
`override def setter[U <: String] = TupleSetter.asSubSetter[String, U](TupleSetter.of[String])` taking the implicit in the constructor changes the binary compatibility without a need that I see.

Reply:
oh cool! kk